lib/kitchen/directions/bake_toc.rb

```diff
@@ -112,7 +112,8 @@ def self.li_for_page(page)
               ['os-toc-unit-page', 'intro']
             else
               raise "could not detect which page type class to apply for page.id `#{page.id}`
-               during baking the TOC. The classes on the page are: `#{page.classes}`"
+               during baking the TOC. The classes on the page are: `#{page.classes}`
+               #{page.data_source ? "\n" : nil}#{page.data_source}"
```

Comment:
i like the `CNXML SOURCE: ` prefix that you have in other places -- can it be added here? 🙂